sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/rest_client_models.hpp

```diff
@@ -74,4 +74,235 @@ namespace Azure { namespace Security { namespace KeyVault { namespace Administra
+       *
+       */
+      Azure::Nullable<std::string> StatusDetails;
+      /**
+       * @brief Error encountered, if any, during the full backup operation.
+       *
+       */
+      Azure::Nullable<KeyVaultServiceError> Error;
+      /**
+       * @brief The start time of the backup operation in UTC
+       *
+       */
+      DateTime StartTime;
+      /**
+       * @brief The end time of the backup operation in UTC
+       *
+       */
+      Nullable<DateTime> EndTime;
+      /**
+       * @brief Identifier for the full backup operation.
+       *
+       */
+      std::string JobId;
+      /**
+       * @brief The Azure blob storage container Uri which contains the full backup
+       *
+       */
+      std::string AzureStorageBlobContainerUri;
+    };
+
+    /**
+     * @brief Sas token parameter for backup and restore operations.
+     *
+     */
+    struct SasTokenParameter final
+    {
+      /**
+       * @brief Azure Blob storage container Uri
+       *
+       */
+      std::string StorageResourceUri;
+      /**
+       * @brief The SAS token pointing to an Azure Blob storage container
+       *
+       */
+      Nullable<std::string> Token;
+      /**
+       * @brief Indicates which authentication method should be used. If set to true, Managed HSM
+       * will use the configured user-assigned managed identity to authenticate with Azure Storage.
+       * Otherwise, a SAS token has to be specified.
+       *
+       */
+      Nullable<bool> UseManagedIdentity;
+    };
+
+    /**
+     * @brief Full backup status options.
+     *
+     */
+    struct FullBackupStatusOptions final
+    {
+      /**
+       * @brief Identifier for the full backup operation.
+       *
+       */
+      std::string JobId;
+    };
+
+    /**
+     * @brief Full restore operation.
+     *
+     */
+    struct RestoreOperation final
+    {
+      /**
+       * @brief Status of the restore operation.
+       *
+       */
+      std::string Status;
+      /**
+       * @brief The status details of restore operation.
+       *
+       */
+      Azure::Nullable<std::string> StatusDetails;
+      /**
+       * @brief Error encountered, if any, during the restore operation.
+       *
+       */
+      Azure::Nullable<KeyVaultServiceError> Error;
+      /**
+       * @brief Identifier for the restore operation.
+       *
+       */
+      std::string JobId;
+      /**
+       * @brief The start time of the restore operation.
+       *
+       */
+      DateTime StartTime;
+      /**
+       * @brief The end time of the restore operation.
+       *
+       */
+      Nullable<DateTime> EndTime;
+    };
+
+    /**
+     * @brief Restore operation parameters.
+     *
+     */
+    struct RestoreOperationParameters final
+    {
+      /**
+       * @brief The SAS token pointing to an Azure Blob storage container where the previous
+       * successful full backup was stored
+       *
+       */
+      SasTokenParameter SasTokenParameters;
```

Comment:
Does it make sense to add commentary about the SAS token being a credential which should be treated as sensitive information?

Reply:
done